### PR TITLE
Wrapper for fetching ETH average gas price from etherscan

### PR DESCRIPTION
**Pull request by sarath-hotspot**
https://github.com/polywrap/mini-hacks/issues/33

### CODE REVIEW

File: ethgasprices/src/query/index.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    gasPriceWei = gasPriceWei.substring(1, gasPriceWei.length - 1);
+    gasPriceMap.set(parsedDate, BigInt.fromString(gasPriceWei));
+  }
+
+  return Result.Ok<Map<string, BigInt>, string>(gasPriceMap);
+}
+
+function fetchGasPricesContentFromEtherScan() : Result<string, string> {
+  const response = Http_Query.get({
+    url: " https://etherscan.io/chart/gasprice",
```
Comment (dOrgJelli) [inline ethgasprices/src/query/index.ts:81]: Would be awesome to see if we can pass a filter argument for a specific date (or range) to the API. This way we don't have to download the full CSV, and extract one value from it.

**Review by dOrgJelli — CHANGES_REQUESTED** [review]

File: ethgasprices/src/query/index.ts (hunk 1 above)
Comment (sarath-hotspot) [reply ethgasprices/src/query/index.ts:81]: I see etherscan provides an programmatic access to gas stats. 

We could build polywrap interface for these APIs. 

https://docs.etherscan.io/api-endpoints/gas-tracker#get-daily-average-gas-price

Comment (sarath-hotspot) [reply ethgasprices/src/query/index.ts:81]: But this specific api(get-daily-average-gas-price) is a pro API and needs subscription.

Comment (Niraj-Kamdar) [reply ethgasprices/src/query/index.ts:81]: You can also create a common interface and provide different implementations for that using different APIs.

**Comment by Niraj-Kamdar** [comment]
@sarath-hotspot Are you still interested in completing this?

**Comment by sarath-hotspot** [comment]
> @sarath-hotspot Are you still interested in completing this?

Yes. I am taking summer break, will resume in September.

**Comment by dOrgJelli** [comment]
Closing this due to inactivity.

**Pull request closed without merging**